src/Query.jl

```diff
@@ -0,0 +1,195 @@
+module Query
+
+export From, Where, Select, SimpleQueryFormatter, NamedQueryFormatter, DFQueryFormatter
+
+using ..ACSetInterface, ..Schemas
+using MLStyle
+using DataFrames: DataFrame
+
+to_name(x) = x
+to_name(x::Pair) = Symbol("$(x.second)$(x.first)")
+to_name(x::Val{T}) where T = Symbol("Val_$T")
+
+# for iterating over something that might be a singleton
+function iterable(x::T) where T
+  S = T <: AbstractVector ? eltype(T) : T
+  [S[]; x]
+end
+
+function Base.get(acset::ACSet, select::Symbol, idx=Colon(); schema=acset_schema(acset))
+  val = select ∈ objects(schema) ? parts(acset, select) : subpart(acset, select)
+  val[idx]
+end
+
+function Base.get(acset, selects::Vector{Symbol}, idx=Colon(); kwargs...)
+  zip(get.(Ref(acset), selects, idx; kwargs...)...)
+end
+
+abstract type AbstractCondition end
+
+struct WhereCondition <: AbstractCondition
+  lhs
+  op::Function
+  rhs
+end
+
+struct AndWhere <: AbstractCondition
+  conds::Vector{<:AbstractCondition}
+  AndWhere(conds::Vector{<:AbstractCondition}) = new(conds)
+  AndWhere(a::AndWhere, b) = AndWhere(a.conds, b)
```

Comment:
the constructors which don't call `new` are usually outside of the `struct` definition. (also there should be a test for them)